bears/general/IndentationBear.py

```diff
@@ -66,6 +66,7 @@ def run(self,
             Full path of external directory containing the coalang
             file for language.
         """
+        ranges = {}
```

Comment:
Is this really an improvement?

60 additions 56 deletions doesn't look like you are saving much code if any at all, and it looks a little less clear what `ranges` are, opposing to `encapsulators`, `comments` etc. 

So what is the value you are adding here? making the algorithm more extensible? or just simplifying things? I'm not convinced just yet.